lib/WP_Auth0_Id_Token_Validator.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Contains WP_Auth0_Id_Token_Validator.
+ *
+ * @package WP-Auth0
+ *
+ * @since 3.11.0
+ */
+
+/**
+ * Class WP_Auth0_Id_Token_Validator.
+ * Wrapper around JWT::decode() to do additional checks and enforce defaults.
+ */
+class WP_Auth0_Id_Token_Validator {
+
+	/**
+	 * ID token to decode.
+	 *
+	 * @var string
+	 */
+	protected $id_token;
+
+	/**
+	 * Key to use to verify the signature.
+	 *
+	 * @var bool|string
+	 */
+	protected $key;
```

Comment:
Can we call this secret to better indicate it is sensitive?

Reply:
@damieng - This may or may not be secret. In the case of HS256 it is but for RS256 it's a public key from the JWKS.

Reply:
Looks like it's only assigned from   

>  		$this->key       = $opts->get_client_secret_as_key();

Reply:
Bad legacy method name. It gets the actual client secret from the database if the token algorithm is set to `HS256` and gets the public key from the JWKS if it's set to `RS256`.